### DISCUSSION TITLE
What's the correct workflow to use this repo?

### DISCUSSION THREAD

**Discussion by mendax1234** [academicpages/academicpages.github.io#1938] [Q&A]
I want to know what's the correct workflow to use this repo, since I see that sometimes there are several updates to this academicspage. So, the correct one is to fork this repo and fetch from the source repo regularly to make sure the new functions is added. However, I think that using this method may cause a potential conflict problem. So, may I ask what's the correct workflow to use this awesome project, or is there a way for me to get the latest updates of this repo as well as don't need to resolve the conflicts.

Thank you very much!

**Comment by rjzupkoii** [comment]
This has been a bit of a long standing issue with the template (see Issue #7) and while I have plans to try and address it, one of major pain points is that some files need to be modified in the forked repositories (i.e., `_config.yml`) but those changes get clobbered when you fetch from the source repository. streamlining that process is going to take some time which is why it's not really on the horizon until the [v0.9 Milestone](https://github.com/academicpages/academicpages.github.io/milestone/3).

In the meantime there are really only two viable workflows that people tend to use:
1. Fork the repository template and then detach it so you stop getting notifications. This means that you will have to keep an eye on all maintenance issues, but if you are making extensive changes to the template then it might also make sense.
2. Back-up the files that you changed, fetch the changes, and then restore the files that got clobbered. If the majority of your edits are limited to the `_config.yml` then this is a pretty reasonable reasonable route since fetching from the source should not delete files that you add (e.g., by adding a new publication to `_publications`).

**Reply by mendax1234** [reply]
Thanks for your reply! So for now, only by forking this repo can I get the latest update? However, I see that there is another option called "use the template", will this also allow me to get the latest update from the parent repo?

Thanks!

**Reply by rjzupkoii** [reply]
@mendax1234  Right now the best approach is for via the "Use the template" option (this avoids copying extra material that is only really useful if you wish to help maintain the template), and keep an eye out for any updates. Most users only edit `_config.yml` and add new files, so updating shouldn't undo much progress.
